apps/coordinator/src/components/Wallet/TransactionsTab/FeeBumping/utils.ts

```diff
@@ -0,0 +1,238 @@
+
+// =============================================================================
+// HELPER UTILITY FUNCTIONS FOR FORMATTING AND VALIDATION
+// =============================================================================
+
+/**
+ * Formats a fee amount in satoshis to a human-readable string with locale-aware formatting
+ *
+ * This function handles large numbers gracefully and provides consistent formatting
+ * across different locales. It's particularly useful for displaying transaction fees
+ * in user interfaces.
+ *
+ * @param feeInSatoshis - The fee amount in satoshis (can be string or number)
+ * @param includeSuffix - Whether to append " sats" to the formatted string
+ * @returns Formatted fee string with thousands separators (e.g., "1,234 sats")
+ *
+ * @example
+ * formatFee(1234567) // "1,234,567 sats"
+ * formatFee("5000", false) // "5,000"
+ * formatFee("invalid") // "0 sats"
+ */
+export const formatFee = (
+  feeInSatoshis: string,
+  includeSuffix = true,
+): string => {
+  // Parse the fee as a number
+  const fee = parseInt(feeInSatoshis, 10);
+
+  if (isNaN(fee)) return "0" + (includeSuffix ? " sats" : "");
+
+  // Format with thousands separators
+  const formattedFee = fee.toLocaleString();
+
+  return formattedFee + (includeSuffix ? " sats" : "");
+};
+
+/**
+ * Comprehensive validation of transaction inputs required for fee analysis
+ *
+ * This function performs critical validation to ensure the fee bumping process
+ * has all necessary data. It uses structured error reporting to help with
+ * debugging and user feedback.
+ *
+ * @param txHex - Raw transaction hex string from the blockchain
+ * @param fee - Current transaction fee in satoshis
+ * @param availableUtxos - UTXOs available for fee bumping operations
+ * @throws {Error} When validation fails with descriptive error messages
+ *
+ * @internal This is a private validation function used internally
+ */
+const validateTransactionInputs = (
+  txHex: string,
+  fee: number,
+  availableUtxos: FeeUTXO[],
+): void => {
+  if (!txHex) {
+    throw new Error("Transaction hex must be a string");
+  }
+
+  if (isNaN(fee) || fee < 0) {
+    console.warn("Invalid fee provided, using 0");
+  }
+
+  if (!availableUtxos?.length) {
+    throw new Error("No UTXOs available for fee bumping");
+  }
+};
+
+/**
+ * Identifies the change output in a transaction by analyzing output addresses
+ * and wallet data
+ *
+ * This function uses multiple heuristics to identify which output is the change:
+ * 1. Matches against known wallet addresses
+ * 2. Checks BIP32 path patterns (change addresses use path m/1/*)
+ * 3. Position in outputs (change is often the last output)
+ *
+ * @param transaction - The transaction object
+ * @param walletState - The wallet state containing addresses
+ * @returns Index of the change output or undefined if not found
+ *
+ * @see https://en.bitcoin.it/wiki/Privacy#Change_address_detection
+ */
+export const getChangeOutputIndex = (
```

Comment:
Let's put a pin in this for the later PRs to see if we can avoid this altogether as a util. It could probably be done as a hook:

```
const changeOutputIndex = useGetChangeOutputIndex(tx)
```

where `useGetChangeOutputIndex` can just take a t and get the addresses directly from state. Not a blocker but want to make sure we come back for cleanup if we can.